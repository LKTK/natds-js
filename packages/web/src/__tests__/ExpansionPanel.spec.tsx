import React from "react";
import { expect } from "chai";
import renderer from "react-test-renderer";

import {
  ExpansionPanel,
  ExpansionPanelActions,
  ExpansionPanelDetails,
  ExpansionPanelSummary,
} from "..";

describe("ExpansionPanel component", () => {
  test("should match snapshot - Simple children", () => {
    const component = renderer.create(<ExpansionPanel><div>content</div></ExpansionPanel>).toJSON();

    expect(component).matchSnapshot("ExpansionPanel with simple children");
  });

  test("should match snapshot - ExpansionPanelActions children", () => {
    const component = renderer.create(<ExpansionPanel><ExpansionPanelActions>content</ExpansionPanelActions></ExpansionPanel>).toJSON();

    expect(component).matchSnapshot("ExpansionPanel with ExpansionPanelActions");
  });

  test("should match snapshot - ExpansionPanelDetails children", () => {
    const component = renderer.create(<ExpansionPanel><ExpansionPanelDetails>content</ExpansionPanelDetails></ExpansionPanel>).toJSON();

    expect(component).matchSnapshot("ExpansionPanel with ExpansionPanelDetails");
  });

  test("should match snapshot - ExpansionPanelSummary children", () => {
    const component = renderer.create(<ExpansionPanel><ExpansionPanelSummary>content</ExpansionPanelSummary></ExpansionPanel>).toJSON();

    expect(component).matchSnapshot("ExpansionPanel with ExpansionPanelSummary");
  });

  test("should match snapshot - ExpansionPanel complete", () => {
    const component = renderer.create(<ExpansionPanel defaultExpanded>
      <ExpansionPanelSummary>Summary</ExpansionPanelSummary>
      <ExpansionPanelDetails>Details</ExpansionPanelDetails>
      <ExpansionPanelActions>Actions</ExpansionPanelActions>
    </ExpansionPanel>).toJSON();

    expect(component).matchSnapshot("ExpansionPanel with complete example");
  });
});
