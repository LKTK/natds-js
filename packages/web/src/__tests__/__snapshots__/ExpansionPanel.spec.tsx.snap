// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpansionPanel component should match snapshot - ExpansionPanel complete: ExpansionPanel with complete example 1`] = `
<div
  className="MuiPaper-root MuiExpansionPanel-root Mui-expanded MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-disabled={false}
    aria-expanded={true}
    className="MuiButtonBase-root MuiExpansionPanelSummary-root Mui-expanded"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiExpansionPanelSummary-content Mui-expanded"
    >
      Summary
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="MuiExpansionPanelDetails-root"
          >
            Details
          </div>
          <div
            className="MuiExpansionPanelActions-root MuiExpansionPanelActions-spacing"
          >
            Actions
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ExpansionPanel component should match snapshot - ExpansionPanelActions children: ExpansionPanel with ExpansionPanelActions 1`] = `
<div
  className="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiExpansionPanelActions-root MuiExpansionPanelActions-spacing"
  >
    content
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ExpansionPanel component should match snapshot - ExpansionPanelDetails children: ExpansionPanel with ExpansionPanelDetails 1`] = `
<div
  className="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiExpansionPanelDetails-root"
  >
    content
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ExpansionPanel component should match snapshot - ExpansionPanelSummary children: ExpansionPanel with ExpansionPanelSummary 1`] = `
<div
  className="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="MuiButtonBase-root MuiExpansionPanelSummary-root"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiExpansionPanelSummary-content"
    >
      content
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ExpansionPanel component should match snapshot - Simple children: ExpansionPanel with simple children 1`] = `
<div
  className="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div>
    content
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        />
      </div>
    </div>
  </div>
</div>
`;
